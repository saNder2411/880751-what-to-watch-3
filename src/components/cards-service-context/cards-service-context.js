import {createContext} from 'react';

const {
  Provider: CardsServiceProvider,
  Consumer: CardsServiceConsumer
} = createContext();

export {CardsServiceProvider, CardsServiceConsumer};
