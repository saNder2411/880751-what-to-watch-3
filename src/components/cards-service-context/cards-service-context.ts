import * as React from 'react';

const {
  Provider: CardsServiceProvider,
  Consumer: CardsServiceConsumer
} = React.createContext(void 0);

export {CardsServiceProvider, CardsServiceConsumer};
