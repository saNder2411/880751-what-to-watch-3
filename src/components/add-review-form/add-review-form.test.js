import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter, Switch, Route} from 'react-router-dom';
import CardsService from '../../services/cards-service';
import {CardsServiceProvider} from '../cards-service-context/cards-service-context';
import createAPI from '../../api';
import thunk from 'redux-thunk';

import AddReviewForm from './add-review-form';
import withAddReviewFormState from '../../hocs/with-add-review-form-state/with-add-review-form-state';

const API = createAPI(() => {});
const cardsService = new CardsService(API);
const mockStore = configureStore([thunk]);


const store = mockStore({
  appState: {
    screen: `MAIN`,
    selectedCardId: 1,
  },
  reviews: {
    reviewsData: [],
    reviewsLoading: false,
    reviewsError: null,
  }
});

const WrappedAddReviewForm = withAddReviewFormState(AddReviewForm);


it(`Should AddReviewForm render correctly`, () => {
  const markup = renderer
    .create(
        <Provider store={store}>
          <CardsServiceProvider value={cardsService}>
            <BrowserRouter>
              <Switch>
                <Route path='/'>
                  <WrappedAddReviewForm />
                </Route>
              </Switch>
            </BrowserRouter>
          </CardsServiceProvider>
        </Provider>
    )
    .toJSON();

  expect(markup).toMatchSnapshot();
});
