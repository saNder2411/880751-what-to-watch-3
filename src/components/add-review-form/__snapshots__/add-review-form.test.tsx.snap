// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should AddReviewForm render correctly 1`] = `
<div
  className="add-review"
>
  <form
    action=""
    className="add-review__form"
    onSubmit={[Function]}
  >
    <div
      className="rating"
    >
      <div
        className="rating__stars"
      >
        <input
          checked={false}
          className="rating__input"
          id="star-1"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={1}
        />
        <label
          className="rating__label"
          htmlFor="star-1"
        >
          Rating 
          1
        </label>
        <input
          checked={false}
          className="rating__input"
          id="star-2"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={2}
        />
        <label
          className="rating__label"
          htmlFor="star-2"
        >
          Rating 
          2
        </label>
        <input
          checked={false}
          className="rating__input"
          id="star-3"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={3}
        />
        <label
          className="rating__label"
          htmlFor="star-3"
        >
          Rating 
          3
        </label>
        <input
          checked={false}
          className="rating__input"
          id="star-4"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={4}
        />
        <label
          className="rating__label"
          htmlFor="star-4"
        >
          Rating 
          4
        </label>
        <input
          checked={false}
          className="rating__input"
          id="star-5"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={5}
        />
        <label
          className="rating__label"
          htmlFor="star-5"
        >
          Rating 
          5
        </label>
      </div>
    </div>
    <div
      className="sign-in__message"
      style={
        Object {
          "color": "#866866",
        }
      }
    >
      <p>
        Please enter a valid email address
      </p>
    </div>
    <div
      className="add-review__text"
      style={
        Object {
          "boxShadow": "0 0 8px 4px #D36987",
        }
      }
    >
      <textarea
        className="add-review__textarea"
        id="review-text"
        name="review-text"
        onChange={[Function]}
        placeholder="Review text"
        value=""
      />
      <div
        className="add-review__submit"
      >
        <button
          className="add-review__btn"
          disabled={true}
          type="submit"
        >
          Post
        </button>
      </div>
    </div>
  </form>
</div>
`;
